src/viam/sdk/registry/registry.hpp

```diff
@@ -11,84 +11,54 @@
 #include <grpcpp/impl/service_type.h>
 #include <grpcpp/server.h>
 
-#include <viam/api/robot/v1/robot.pb.h>
-
 #include <viam/sdk/config/resource.hpp>
 #include <viam/sdk/resource/resource.hpp>
 #include <viam/sdk/resource/resource_api.hpp>
 #include <viam/sdk/resource/resource_manager.hpp>
 #include <viam/sdk/resource/resource_server_base.hpp>
+#include <viam/sdk/rpc/server.hpp>
 
 namespace viam {
 namespace sdk {
 
 // TODO(RSDK-3030): instead of std::functions, consider making these functions
 // virtual
 // TODO(RSDK-3030): one class per header
-/// @class ResourceRegistration registry.hpp "registry/registry.hpp"
-/// @brief Defines registered `Resource` creation functionality.
-class ResourceRegistration {
+class ResourceServerRegistration {
    public:
-    virtual ~ResourceRegistration();
-
-    /// @brief Add `Reconfigure` functionality to a resource.
-    std::function<std::shared_ptr<Resource>(std::shared_ptr<Resource>, Name)> create_reconfigurable;
```

Comment:
This wasn't doing anything and the (theoretical) semantics of it were confusing. So I got rid of it. Per RSDK-6484, we should just use a `Reconfigurable` base class instead.